File: cpp/56c_1619B_SquaresAndCubes.cpp
#include<bits/stdc++.h>

unsigned int solve(int n){
    unsigned int sq = (sqrt(n)), cube = (cbrt(n));
    unsigned int sixth = (sqrt(cbrt(n)));
    //std::cout<<sq<<" "<<cube<<" "<<sixth<<"\n";
    return (sq + cube - sixth);
}

int main(){
    int t, n;
    std::vector<unsigned int> ans;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>n;
        ans.push_back(solve(n));
    }
    for (auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}

File: python/56c_1619B_SquaresAndCubes.py
from math import sqrt, cbrt

def solve(n):
    return int(sqrt(n)) + int(cbrt(n)) - int(sqrt(cbrt(n)))

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        m = int(input())
        ans.append(solve(m))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()